tasks/db_deploy/test/unit_tests/test_migrate_db.py

```diff
@@ -35,9 +37,9 @@ def tearDown(self):
         """
         self.config = None
 
-    @patch("migrate_db.migrate_versions_1_to_2")
-    @patch("migrate_db.migrate_versions_2_to_3")
-    @patch("migrate_db.migrate_versions_3_to_4")
+    @patch("migrations.migrate_versions_1_to_2.migrate_db_v2.migrate_versions_1_to_2")
+    @patch("migrations.migrate_versions_2_to_3.migrate_db_v3.migrate_versions_2_to_3")
+    @patch("migrations.migrate_versions_3_to_4.migrate_db_v4.migrate_versions_3_to_4")
     def test_perform_migration_happy_path(
             self,
             mock_migrate_v3_to_v4: MagicMock,
```

Comment:
Getting one unit test error overall.

getting error in this test `test_perform_migration_happy_path`.

not sure what is missing. looks like it's not able to create the connection from shared_db library

```
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not translate host name "aws.postgresrds.host" to address: nodename nor servname provided, or not known
```

Details of the error:


```
dsn = 'host=aws.postgresrds.host user=admin password=admin123 port=5432 dbname=user_db', connection_factory = None, cursor_factory = None
kwargs = {'database': 'user_db', 'host': 'aws.postgresrds.host', 'password': 'admin123', 'port': 5432, ...}, kwasync = {}

    def connect(dsn=None, connection_factory=None, cursor_factory=None, **kwargs):
        """
        Create a new database connection.
    
        The connection parameters can be specified as a string:
    
            conn = psycopg2.connect("dbname=test user=postgres password=secret")
    
        or using a set of keyword arguments:
    
            conn = psycopg2.connect(database="test", user="postgres", password="secret")
    
        Or as a mix of both. The basic connection parameters are:
    
        - *dbname*: the database name
        - *database*: the database name (only as keyword argument)
        - *user*: user name used to authenticate
        - *password*: password used to authenticate
        - *host*: database host address (defaults to UNIX socket if not provided)
        - *port*: connection port number (defaults to 5432 if not provided)
    
        Using the *connection_factory* parameter a different class or connections
        factory can be specified. It should be a callable object taking a dsn
        argument.
    
        Using the *cursor_factory* parameter, a new default cursor factory will be
        used by cursor().
    
        Using *async*=True an asynchronous connection will be created. *async_* is
        a valid alias (for Python versions where ``async`` is a keyword).
    
        Any other keyword parameter will be passed to the underlying client
        library: the list of supported parameters depends on the library version.
    
        """
        kwasync = {}
        if 'async' in kwargs:
            kwasync['async'] = kwargs.pop('async')
        if 'async_' in kwargs:
            kwasync['async_'] = kwargs.pop('async_')
    
        if dsn is None and not kwargs:
            raise TypeError('missing dsn and no parameters')
    
        dsn = _ext.make_dsn(dsn, **kwargs)
>       conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
E       sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not translate host name "aws.postgresrds.host" to address: nodename nor servname provided, or not known
E       
E       (Background on this error at: http://sqlalche.me/e/14/e3q8)

venv/lib/python3.9/site-packages/psycopg2/__init__.py:127: OperationalError
------------------------------------------------------------------------ Captured stderr call ------------------------------------------------------------------------
{"message": "Creating admin user connection object.", "sender": "unknown", "version": "unknown", "timestamp": "2022-02-03T16:12:40.633712", "level": "debug"}
{"message": "Database set to user_db for the connection.", "sender": "unknown", "version": "unknown", "timestamp": "2022-02-03T16:12:40.633823", "level": "debug"}
{"message": "Creating URL object to connect to the database.", "sender": "unknown", "version": "unknown", "timestamp": "2022-02-03T16:12:40.633897", "level": "debug"}
```

Reply:
I think your Mock patches for the test are incorrect. You should be patching the calls within migrate_db not the base objects. Try changing your patches to be
```
@patch("migrations.migrate_db.migrate_versions_1_to_2")
@patch("migrations.migrate_db.migrate_versions_2_to_3")
@patch("migrations.migrate_db.migrate_versions_3_to_4")
```

Reply:
Sounds like your mocks aren't catching things correctly, but without a stack trace that's all I can infer.

Reply:
tried what Brad suggested and that seems to fix it. 

I wasn't patching the calls from migrate_db but from the individual migration folders which is why it showed error